httpserver/src/main/kotlin/tbdex/sdk/httpserver/TbdexHttpServer.kt

```diff
@@ -134,36 +134,74 @@ class TbdexHttpServer(val config: TbdexHttpServerConfig) {
           getExchange(
             call,
             exchangesApi,
-            getCallbacks.getOrDefault("exchange", null),
+            callbacks.getOrDefault("exchange", null) as GetExchangeCallback?,
             pfiDid
           )
         }
       }
 
       get("/offerings") {
-        getOfferings(call, offeringsApi,  getCallbacks.getOrDefault("offerings", null))
+        getOfferings(
+          call,
+          offeringsApi,
+          callbacks.getOrDefault("offerings", null) as GetOfferingsCallback?
+        )
       }
     }
   }
 
   /**
-   * Adds a submit callback for a specific message kind (RFQ, order, close, etc.).
+   * Adds a GetOfferingsCallback for handling requests for offerings.
+   *
+   * @param callback GetOfferingsCallback function to be registered
+   */
+  fun getOfferings(callback: GetOfferingsCallback) {
+    this.callbacks["offerings"] = callback
```

Comment:
can we offer these string literals as an enum or something to avoid any errors and expose them to the client if needed?

Reply:
now using Callback class and variables housed inside https://github.com/TBD54566975/tbdex-kt/pull/218/commits/7d819ff3f7d79697cf1a80cc87ebdb459c2a8af1